firestore/src/jni/compare.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "firestore/src/android/exception_android.h"
+#include "firestore/src/android/wrapper.h"
+#include "firestore/src/jni/env.h"
+#include "firestore/src/jni/object.h"
+
+/**
+ * Returns the result of `Object::Equals` for the given `lhs` and `rhs` in the
+ * current Firestore `jni::Env`.
+ */
+template <typename T>
+bool CompareJni(const T& lhs, const T& rhs) {
```

Comment:
Ultranit: I think this function should be named `EqualityCompareJni` for consistency with the other one. Sorry I missed that in my suggestions.

Reply:
Done. It also turns out that it would have been better to place this function in the proper namespace (the Android build was not successful as it was looking for `firebase::firestore::*` for Env, Object, and GlobalUnhandledExceptionHandler.)